foreign_cc/cmake.bzl

```diff
@@ -218,6 +218,8 @@ def _create_configure_script(configureParameters):
         for arg in ctx.attr.build_args
     ])
 
+    user_env = {key: ctx.expand_location(value, data) for (key, value) in ctx.attr.env}
```

Comment:
I think this line might be causing these errors:

```
ERROR: /build/lib/z/BUILD.bazel:6:19: in configure_make rule //lib/z:z: label '//lib/python:runtime' in $(locations) expression is not a declared prerequisite of this rule
```

Or maybe I'm doing something wrong, but just changing `configure_env_vars` in my `//lib/z` to `//vars` breaks the build with `--cpu=wasm32`. I am passing these env vars:

```
"PATH": "${EXT_BUILD_ROOT}/$(execpaths //lib/python:runtime)/bin:${PATH}",
"PYTHONHOME": "${EXT_BUILD_ROOT}/$(execpaths //lib/python:runtime)",
```

Reply:
Or rather, the issue I'm facing is coming from `configure.bzl` but the idea is the same. How come the locations were expanded earlier, even though you did not call `ctx.expand_location()` on the `env_var` attribute earlier?

Reply:
Can you share the build script that's generated in the before and after?

Reply:
Doesn't `execpaths` return multiple files? What is `${EXT_BUILD_ROOT}/$(execpaths //lib/python:runtime)/bin:${PATH}` expected to render into?

Reply:
Yes, `execpaths` returns multiple paths, but in my case `//lib/python:runtime` is a filegroup that selects the `gen_dir` of the Python target, so it only contains one file (the directory being built).

```bazel
configure_make(
    name = "python",
    …
)

filegroup(
    name = "runtime",
    srcs = [":python"],
    output_group = "gen_dir",
)
```

So this way `//lib/python:runtime` expands to just one value which contains the binaries, libraries, header files and the `$PYTHONHOME`.

Reply:
Somehow I'm still getting the same error, even after applying 7ae1ea4. This line is for example now failing:

https://github.com/attilaolah/wasm/blob/6f2bac587be981264e90cf4addbbc9bbb7454006/lib/gmp/BUILD.bazel#L15

With the error:

```
ERROR: /build/lib/gmp/BUILD.bazel:7:19: in configure_make rule //lib/gmp:gmp: label '//lib/m4:runtime' in $(locations) expression is not a declared prerequisite of this rule
```

Reply:
Oh, I think this was a dumb oversight on my part 😅 

Try now and see the last commit.

Reply:
Now it works! 🎉

But you fixed it so quickly, that I switched to using a `local_repository()` in the meantime to try and debug the issue. Then I added your fork as a remote and checked it out locally, but by the time I did that, you already committed the fix, so then I had my build passing with a local repo, but failing when patched in this pull request, and so I was scratching my head trying to figure out how the tow was different. 😀

Anyhow, thanks for fixing, I think this should be good to go but I'll quickly do a final build just to double-check.

Reply:
Let me know when you've double checked. Then I'd ask that you review and approve the changes here and we can get this closed out 😄